### ISSUE TITLE
CommunityToolkit.Mvvm source generators fail on Visual Studio 17.4.0 Preview 6 and 17.5.0 Preview 1

### ISSUE THREAD

**Issue by nxtn-staged** [CommunityToolkit/dotnet#493]
> **Note**: this is an external issue in Roslyn, see: https://github.com/dotnet/roslyn/issues/67123.

### Describe the bug

I recently updated Visual Studio to 17.5.0 Preview 1.0 and my project starts to fail to build.

```
1>CSC : warning CS8785: Generator 'INotifyPropertyChangedGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'MissingMethodException' with message 'Method not found: 'System.ReadOnlySpan`1<System.__Canon> System.Collections.Immutable.ImmutableArray`1.AsSpan()'.'
1>CSC : warning CS8785: Generator 'ObservablePropertyGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'MissingMethodException' with message 'Method not found: 'System.ReadOnlySpan`1<System.__Canon> System.Collections.Immutable.ImmutableArray`1.AsSpan()'.'
```

### Steps to reproduce

1. Add project files

```csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="CommunityToolkit.Mvvm" Version="8.1.0-preview1" />
  </ItemGroup>
  
</Project>
```

```cs
using CommunityToolkit.Mvvm.ComponentModel;

[INotifyPropertyChanged]
partial class Foo
{
    [ObservableProperty]
    private int bar;
}

static class Program
{
    static void Main()
    {
        new Foo().Bar = 42;
    }
}
```

2. Build in Visual Studio

### Expected behavior

It works.

### IDE and version

VS 2022 Preview

### Nuget packages

- [ ] CommunityToolkit.Common
- [ ] CommunityToolkit.Diagnostics
- [ ] CommunityToolkit.HighPerformance
- [X] CommunityToolkit.Mvvm (aka MVVM Toolkit)

### Nuget package version(s)

8.1.0-preview1

### Additional context

I can't always reproduce this. I'm also not sure if it's a problem in this project, but my other source generator dependencies are not affected.

### Help us help you

Yes, but only if others can assist

**Comment by Sergio0694** [comment]
Mmh this is weird. Just to clarify:
- Does this also repro on VS 17.4?
- Does this repro on 8.0.0 as well, or just on 8.1.0-Preview1?

**Comment by nxtn-staged** [comment]
It first reproduced when I updated Visual Studio to the last 17.4.0 Preview version (can't remember which version it was, [release notes](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes-preview) have been cleared) and some package dependencies. Downgrading CommunityToolkit.Mvvm from 8.1.0-Preview1 to 8.0.0 didn't help, and I managed to work around it by downgrading Microsoft.Windows.CsWin32 from 0.2.104-beta to 0.2.63-beta 😕. After updating VS to 17.5.0 Preview 1 it doesn't work anymore.

**Comment by arkadym** [comment]
Im stuck with below errors after upgrade
```
1>...\CommunityToolkit.Maui.Markup.SourceGenerators\CommunityToolkit.Maui.Markup.SourceGenerators.TextAlignmentExtensionsGenerator\EditorTextAlignmentExtensions.g.cs(122,147,122,167): error CS0121: The call is ambiguous between the following methods or properties: 'CommunityToolkit.Maui.Markup.TextAlignmentExtensions_Editor.TextCenterHorizontal(Microsoft.Maui.Controls.Editor)' and 'CommunityToolkit.Maui.Markup.TextAlignmentExtensions_Editor.TextCenterHorizontal(Microsoft.Maui.Controls.Editor)'
1>...\CommunityToolkit.Maui.Markup.SourceGenerators\CommunityToolkit.Maui.Markup.SourceGenerators.TextAlignmentExtensionsGenerator\EntryTextAlignmentExtensions.g.cs(122,145,122,165): error CS0121: The call is ambiguous between the following methods or properties: 'CommunityToolkit.Maui.Markup.TextAlignmentExtensions_Entry.TextCenterHorizontal(Microsoft.Maui.Controls.Entry)' and 'CommunityToolkit.Maui.Markup.TextAlignmentExtensions_Entry.TextCenterHorizontal(Microsoft.Maui.Controls.Entry)'
1>...\CommunityToolkit.Maui.Markup.SourceGenerators\CommunityToolkit.Maui.Markup.SourceGenerators.TextAlignmentExtensionsGenerator\LabelTextAlignmentExtensions.g.cs(122,145,122,165): error CS0121: The call is ambiguous between the following methods or properties: 'CommunityToolkit.Maui.Markup.TextAlignmentExtensions_Label.TextCenterHorizontal(Microsoft.Maui.Controls.Label)' and 'CommunityToolkit.Maui.Markup.TextAlignmentExtensions_Label.TextCenterHorizontal(Microsoft.Maui.Controls.Label)'
1>...\CommunityToolkit.Maui.Markup.SourceGenerators\CommunityToolkit.Maui.Markup.SourceGenerators.TextAlignmentExtensionsGenerator\PickerTextAlignmentExtensions.g.cs(122,147,122,167): error CS0121: The call is ambiguous between the following methods or properties: 'CommunityToolkit.Maui.Markup.TextAlignmentExtensions_Picker.TextCenterHorizontal(Microsoft.Maui.Controls.Picker)' and 'CommunityToolkit.Maui.Markup.TextAlignmentExtensions_Picker.TextCenterHorizontal(Microsoft.Maui.Controls.Picker)'
```

**Comment by huoyaoyuan** [comment]
It reproduces for VS 17.4 stable, for version 8.0.0 and 8.1.0-preview1 both.

**Comment by huoyaoyuan** [comment]
In decompilation, the `System.Collections.Immutable.dll` in .NET Framework version of VS at `MSBuild\Current\bin\Roslyn` does have the method `AsSpan()`. So does the one at `MSBuild\Current\bin`.
I have totally no idea why it can cause `MissingMethodException`.

A workaround is to run `dotnet build` once in command line to use the .NET Core version of compiler, as long as it's not a WinUI project (or the vm project has been separated from UI).

**Comment by Sergio0694** [comment]
Mmh this is weird. I can't repro locally with VS 17.4, but the fact people can with 8.0 stable is worrying.

@sharwell @chsienki do you have any idea what could be causing this? 😅

**Comment by JPhilC** [comment]
Yesterday I was trying to work out why the source generators sometimes worked with WPF (,NET Framework 4.7.1) App and sometimes didn't. This morning I update VS 2022 to 17.4.1 and the source generators have stopped working completely.

Testing with the following code in a class definition.

```
using CommunityToolkit.Mvvm.ComponentModel;

namespace WpfApp4
{
    public class MainWindowModelview: ObservableObject
    {
        [ObservableProperty]
        private string _firstName = "Phil";

        
    }
}
```
The solution builds but when the expected property FirstName is not generated by the source generator.

Steps to reproduce,

Create a new WPF Desktop App (.NET Framework 4.7.1)
Add  nuget V8.1.0 ..Preview 1
Add class MainMenuModelview with above definition.

Build.

**Comment by jansaris** [comment]
On our build environment at the customer we also have this issue.
It's a build-flipper, since it sometime causes the build to fail and sometimes to succeed

We can't see any difference in the generated csproj file when looking at the msbuild binlog file :-(

**Comment by Sergio0694** [comment]
Given that there's a whole bunch of inbox analyzers also failing for the same reason, this seems unrelated from the MVVM Toolkit specifically. Also note that the generators have known issue with non-SDK-style .NET Framework projects. Does the issue still repro for you if you switch to an SDK-style .csproj format?

**Comment by Lightczx** [comment]
I'm using .NET 7, VS2022 17.4.2, SDK style project

Still can reproduce these,@Sergio0694
Everything works fine before I upgrade my `Microsoft.Windows.CsWin32`
``` xml
<PackageReference Include="Microsoft.Windows.CsWin32" Version="0.2.138-beta">
<PackageReference Include="Microsoft.Windows.CsWin32" Version="0.2.164-beta">
```

**Comment by virzak** [comment]
I have the same issue which is also intermittent and I have no idea what triggers it and how it gets resolved.

My error message happens with RelayCommand:
```
5>CSC : warning CS8785: Generator 'RelayCommandGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'MissingMethodException' with message 'Method not found: 'System.ReadOnlySpan`1<!0> System.Collections.Immutable.ImmutableArray`1.AsSpan()'.'
```

There is nothing that seems to be a proven workaround if using VS. Deleting all obj / bin folders does not help. The only thing that helps is switching to VS Code and after a while switching back to VS.

I also use `Microsoft.Windows.CsWin32`, but not sure this has anything to do with this problem.

I have an SDK-style .NET 7 project where this happens.

**Comment by Lightczx** [comment]
@virzak I solved this by downgrade the Cswin32 version, save and close proj, reopen vs ,and then manually upgrade back by editing the csproj file.
After reopen vs, everything works.

**Comment by Sergio0694** [comment]
This is... Uh... Weird. I don't think different generators should/could interfere with one another 🤔

FYI @sharwell

**Comment by Lightczx** [comment]
@Sergio0694 VS17.4.3 has fixed this issue
https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes#17.4.3

**Comment by hymccord** [comment]
I still don't think this is fixed yet. I'm still getting the same error on a personal project of mine.

Installed v17.4.3 in a Windows Sandbox since I normally use the preview versions (and was still seeing the error in 17.5p2)

CommunityToolkit: v8.1.0-preview2
Microsoft.Windows.CsWin32: v0.2.164-beta

```1>CSC : warning CS8785: Generator 'RelayCommandGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'MissingMethodException' with message 'Method not found: 'System.ReadOnlySpan`1<System.__Canon> System.Collections.Immutable.ImmutableArray`1.AsSpan()'.'```

**Comment by Sergio0694** [comment]
Just to clarify:

- Does this only happen in a .NET Framework project or any project?
- Does this only happen if you also reference CsWin32?

**Comment by hymccord** [comment]
I'm putting together a minimal repo now.

**Comment by Lightczx** [comment]
Exactly Microsoft.Windows.CsWin32: v0.2.164-beta.
Lower version like 0.2.138 won't cause any problem.

**Comment by hymccord** [comment]
Only with CsWin32, above version 0.2.138-beta. I'll post and issue over there as well.

Quick repo
```cmd
> mkdir CTRepro
> cd CTRepro
> dotnet new wpf
> dotnet add package CommunityToolkit.Mvvm -v 8.1.0-preview2
> dotnet add package Microsoft.Windows.CsWin32 -v 0.2.164-beta
```

Then add a `.cs` file so a RelayCommand will generate
```cs
using CommunityToolkit.Mvvm.Input;

namespace CTRepro;

internal partial class MainWindowViewModel
{
    [RelayCommand]
    private void Execute()
    { }
}
```

You can do either one of these. MSBuild will at least warn you something went wrong.
```
msbuild /t:Rebuild
```
or 
```
dotnet build
```
What's even worse is that `dotnet build` will say it compiled successfully (**with NO warning**), yet if you inspect the dll with ILSpy or dnSpy, the ExecuteCommand will **NOT** exist.

**Comment by Sergio0694** [comment]
This is... Very weird. I would say I have no idea what's going on. It would seem like CsWin32 is somehow interfering with the toolkit, but even then... How? Like, maybe a .targets that's accidentally interfering with other analyzers? 🤔

Marking this as external for now as it does seem unrelated to the MVVM Toolkit.

**Comment by l619534951** [comment]
The same problem, 
```
Visual Studio: 17.4.3  
Dotnet: 6.0  
CommunityToolkit: 8.0,
```
appears when the cswin32 version is greater than 0.2.63-beta。
cswin32 `0.2.63-beta` Dependencies: System.Memory >= 4.5.4 (System.Memory.dll file version: 4.6.28619.1)
cswin32 `0.2.63-beta later` Dependencies: System.Memory >= 4.5.5 (System.Memory.dll file version: 4.6.31308.1)
**Temporary solution**
**copy** 
`C:\Users\YourUserName\.nuget\packages\microsoft.windows.cswin32\0.2.63-beta\analyzers\cs\System.Memory.dll`
**to**
`C:\Users\YourUserName\.nuget\packages\microsoft.windows.cswin32\0.2.164-beta\analyzers\cs\System.Memory.dll`

@Sergio0694 FYI

**Comment by vitkuz573** [comment]
I have the same problem. I also use CsWin32

CommunityToolkit: v8.1.0-preview2
Microsoft.Windows.CsWin32: v0.2.177-beta

**Comment by sharwell** [comment]
I'm not surprised that CsWin32 is contributing to this. It ships copies of many framework assemblies that may or may not match the ones VS is expecting.
https://github.com/microsoft/CsWin32/blob/933ab8f65d7022b486d1f6cca4fce7a3697f61ad/src/Microsoft.Windows.CsWin32/Microsoft.Windows.CsWin32.nuspec#L51-L58

**Comment by Sergio0694** [comment]
@sharwell Doesn't Roslyn load each analyzer in a separate ALC? I'm confused as to how those can conflict with one another. At the very least, it doesn't seem ideal that an analyzer can break another one and then cause Roslyn to report errors as if the second one was the cause of the issue (eg. see this issue here), as that might cause (like in this case) people to assume it's the fault of the second analyzer, and then go open issues there. For instance, the MVVM Toolkit isn't bundling any additional dependencies, so it seems to me this issue is not its fault, but I'm not sure there's anything we should/could do? 🥲

**Comment by hymccord** [comment]
Roslyn will load each analyzer into an ALC iff it's running under .NET. However, if you are building in Visual Studio then it will use the Full Framework MSBuild (and Roslyn) in `%Program Files%\Microsoft Visual Studio\2022\<Edition>\MSBuild\Current\Bin` which will be .NETFramework based. And probably as you know, you have now entered DLL hell.

It's consistent as to why it succeeds with 'dotnet build` but not VS.

What's weird is that I'm executing `csc.exe` (.NETFramework) directly with the same arguments, and don't get the compilation warning and the resulting dll does have the source generated items.

**Comment by hymccord** [comment]
This one is related. Probably same underlying issue but this one is about how they want the build to fail if you get a CS8785 (treated as error): https://github.com/dotnet/roslyn/issues/65313

**Comment by hymccord** [comment]
Created a specific issue in Roslyn
https://github.com/dotnet/roslyn/issues/66104

I'll create one in CsWin32 as well so they are aware.

**Comment by myblindy** [comment]
I just ran into the exact same issue, a WinUI3 project with the community toolkit as a reference suddenly stops compiling with the same missing method exceptions when adding a reference to `CsWin32`. I guess I'll manually copy the generated files into my project and unreference `CsWin32` for now.

**Comment by myblindy** [comment]
Hm even after removing `CsWin32`, updating the community toolkit to the new `preview2` causes that same error eventually (though not right away). I'll downgrade this as well, it seems to work for now like this.

**Comment by Sergio0694** [comment]
Preview2 has a known issue that causes it to be broken when using it from legacy-style .csproj projects, is that the case for you? I've added a workaround in main, just haven't pushed that to NuGet yet. Feel free to grab the latest preview build from main (from the nightly feed, you can find that in the Wiki for the Windows Community Toolkit, it's the same feed) and let me know if that does the trick! 🙂

See: #540.

**Comment by myblindy** [comment]
> Preview2 has a known issue that causes it to be broken when using it from legacy-style .csproj projects, is that the case for you?

No, it's the new project template, using WinUI3.

**Comment by Sergio0694** [comment]
@myblindy Just for good measure, can you try with 8.1.0 Preview 3 (just published that yesterday) and see if you still repro the issue _without_ CsWin32 as well? Thank you! 🙂

**Comment by myblindy** [comment]
Okay, I updated to preview3 and will let you know how it goes. So far it compiles, but it took a while for it to break earlier.

**Comment by sharwell** [comment]
> Is there an issue for this over in the Roslyn repo? If not it seems like there should be. Being able to be completely broken by other generators doesn't seem ideal, especially if Roslyn reports this one as crashing, leading people to report bugs here.

My general recommendation is don't reference any assemblies from a source generator aside from items provided by Microsoft.CodeAnalysis.CSharp / Microsoft.CodeAnalysis.VisualBasic (the compiler layer). The exception to this is for users who are creating source generators only for use in their own internal projects, where they control all dependencies. For source generators distributed more widely, every additional reference introduces risk proportional to the popularity of that reference.

**Comment by Sergio0694** [comment]
> "My general recommendation is don't reference any assemblies from a source generator aside from items provided by Microsoft.CodeAnalysis.CSharp / Microsoft.CodeAnalysis.VisualBasic"

@sharwell Sure, but there's two big issues I see here:
- If people can still do this and break _other generators_ this seems like a major issue with tooling to me. I shouldn't have my generator break because of someone else, with Visuals Studio then basically blaming my generator and making people assuming that's where the issue here, and then report issues here.
-  The docs in the Roslyn repo about this (I mean the .md file) don't make it clear enough that this should never be done in production for packages that are distributed. I mean if CsWin32 is doing this, and that's an official package from us (MSFT), clearly something has gone wrong 😅

**Comment by mobycorp** [comment]
@Sergio0694 - Why is this closed? This happens regularly in VS 17.4.4 using most of the Community Toolkits... I see it happening mostly from CommunityToolkit.Mvvm v8.1.0 (not preview...)

I have to shutdown VS and restart it and it works for about 20 minutes until I change anything in a file where I have used [ObservableProperty] and [RelayCommand]. It has become untenable to continuously restart VS just to get the code generators to work...  This issue should NOT be closed!!!

**Comment by mobycorp** [comment]
FOLLOW UP: I get error messages like this when change something in a class marked `ObservableObject`. When I exit VS and re-enter, these messages go away because whatever bug is in the code generator gets cleared...

```
Generator 'ObservablePropertyGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentException' with message 'SyntaxTree is not part of the compilation Parameter name: syntaxTree'

Generator 'RelayCommandGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentException' with message 'SyntaxTree is not part of the compilation Parameter name: syntaxTree'
```

I have no idea why someone would give this a thumbs down. These are my legitmate observations.

**Comment by sharwell** [comment]
> If people can still do this and break other generators this seems like a major issue with tooling to me

It's more an issue of limitations of the runtime. Different versions of hosts/runtimes have different nuances, and depending on the project configuration it's hard to say what is or isn't going to work in all cases. The best way I've found for an analyzer to isolate itself from potential external influence is to avoid references beyond those provided by the compiler.

The official position is it's supported to have these references as long as the reference is included in the package with the analyzer. However, since this support is conditioned on a downstream requirement that projects only use analyzers where overlapping dependencies have the same version, it makes things more complex for users.

**Comment by Sergio0694** [comment]
> "The best way I've found for an analyzer to isolate itself from potential external influence is to avoid references beyond those provided by the compiler."

Yes, but _we_ are doing that, and yet if other analyzers don't respect this, users see _our_ analyzer crashing and assume it's an issue on our end, and open issues here. This is what I meant by "tooling issue": the way things are setup, you are completely unable to avoid crashing even if you're doing everything wrong, and worst of all, users are mistakenly led to believe that the issue is in your analyzer, and not where it actually is (ie. in some other analyzer that's bundling stuff). At the very least it's inconvenient 🥲

**Comment by sharwell** [comment]
Here's the PR to fix cases where the conflict occurs for an assembly provided by the host:
https://github.com/dotnet/roslyn/pull/66492

**Comment by Sergio0694** [comment]
Yup yup, @jaredpar was just explaining on Discord how that whole thing was being addressed in his various PRs 😄
To be clear, I do understand this is all an extremely complicated system and that this arguably a niche scenario, and I do appreciate the work all you folks are putting into addressing this. I'm glad at least you were able to figure out what was going on 🙂

**Comment by MichaelShapiro** [comment]
I too appreciate all the work folks are doing to address this. It's just frustrating that on my end I am totally unable to even isolate the culprit that's causing all this. All I see is that things (VS 2022 17.4.4, MAUI, net7 droid and ios, CommunityToolkit.Mvvm 8, CommunityToolkit.Maui 4, AppCenter 5.0.1) worked perfectly ok.. until I updated to CommunityToolkit.Mvvm 8,0.1. Now, the moment I change or add method attributed with RelayCommand  in a viewmodel I get this warning (that comes with multiple other errors):

`Warning	CS8785	Generator 'RelayCommandGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentException' with message 'The hintName 'NAMEOFTHEPROJECT.ViewModel.RegistrationViewModel.Register.g.cs' of the added source file must be unique within a generator. Parameter name: hintName'`

If I rollback to CommunityToolkit.Mvvm 8.0.0 - things work as expected. I created a new project with the same Nuget packages and tried to consistently replicate the errors. I couldn't. Sometimes it works, and sometimes it just fails with the above mentioned warning.

Any suggestions what would be the best course in trying to isolate the offending analyzer or whatever it might be?

**Comment by Sergio0694** [comment]
Yeah that seems similar/same as #343?

**Comment by HavenDV** [comment]
```
3>CSC : warning CS8785: Generator 'ObservablePropertyGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'MissingMethodException' with message 'Method not found: 'System.ReadOnlySpan`1<System.__Canon> System.Collections.Immutable.ImmutableArray`1.AsSpan()'.'
3>CSC : warning CS8785: Generator 'RelayCommandGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'MissingMethodException' with message 'Method not found: 'System.ReadOnlySpan`1<System.__Canon> System.Collections.Immutable.ImmutableArray`1.AsSpan()'.'
```
I am getting these same exceptions in VS 17.4.4 and VS 17.5.0 preview 5
It's important to note that this only happens on Windows 11 ARM64 and Visual Studio ARM64.
Rider 2022.3.1 for ARM64 builds the same code correctly (enabled EnableWindowsTargeting).

**Comment by mobycorp** [comment]
I concur with the edit done to @HavenDV's update. I, too, am seeing it on VS 17.4.4 x64.

**Comment by michael-hawker** [comment]
@HavenDV @mobycorp what version of `CsWin32` are you using, can you try the latest [`v0.2.188-beta`](https://github.com/microsoft/CsWin32/releases/tag/v0.2.188-beta) is supposed to have a fix.

**Comment by HavenDV** [comment]
I updated and everything works. It is worth noting that, apparently, the very presence of CsWin32 in the solution brings problems, and not just its presence in a particular project.

**Comment by michael-hawker** [comment]
@HavenDV yes, if it's referenced anywhere. My understanding of the problem is that CsWin32 was shipping dlls of dependencies that effectively replaced or got confused with the ones in VS, which then the MVVM Toolkit would try and use and would be mismatched from the expected ones in VS, and therefore the generators would fail.

**Comment by polymorphicshade** [comment]
Just chiming in to also report the issue. This issue *significantly* slows down development time for both my professional and personal projects. 

My workflow includes constantly adding/removing/modifying properties and commands, and I have to restart Visual Studio once every few minutes.

**Comment by Sergio0694** [comment]
This doesn't sound normal at all, and I also cannot replicate the issue. Can you share more about your setup, does this happen on the latest VS 17.5 Preview build too? Have you also opened an issue in the Roslyn repo? This doesn't really seem related to the MVVM Toolkit per se, so I'm not really sure there's much we can do from here, unfortunately.

**Comment by virzak** [comment]
This issue has completely gone away for me when I either removed CsWin32 or upgraded it to latest.

**Comment by polymorphicshade** [comment]
@Sergio0694 @virzak 
After reading through things again more carefully, I understand the issue better. Looks like I need to experiment with what you suggested.

I'm using VS 17.4.4 with default installation settings. I'll try tinkering with VS preview and and CsWin32.

**Comment by michael-hawker** [comment]
If folks are using CsWin32, make sure you're on the latest version. There were known issues with older versions. Though maybe it also hasn't been fully resolved either?

**Comment by mobycorp** [comment]
I did not do an specific install for CsWin32. Unless the NuGet installer
installed it, I don't know. What should I do to find it?

**Comment by puszczu** [comment]
> Just chiming in to also report the issue. This issue _significantly_ slows down development time for both my professional and personal projects.
> 
> My workflow includes constantly adding/removing/modifying properties and commands, and I have to restart Visual Studio once every few minutes so the toolkit will start working again.

same issue here, VS keeps spamming me false 'missing property' error all the time, compiling code runs fine, only the whole screen full of red lines on the scroll indicating errors

**Comment by lfhenrl** [comment]
Any news? It is impossibel to work with.

**Comment by hymccord** [comment]
> Any news? It is impossibel to work with.

Are you on the latest version? Using CsWin32? Make sure you are on the latest of that was well. 

If you have other packages that aren’t those, then they may be packaging their NuGets improperly.

**Comment by lfhenrl** [comment]
Sorry but what is CsWin32, Is it a NuGet package I need to install? have seek my hole C drive without any finding.

**Comment by hymccord** [comment]
If you read the previous comments, most of the errors that were mentioned were coupled with the fact users also had the `CsWin32` NuGet package [[link](https://github.com/microsoft/CsWin32)].

Do you have other NuGet packages that are providing Roslyn analyzers/source generators?

**Comment by lfhenrl** [comment]
Thanks for your time.
It is a small POC WinUi 3 project using those packages

    <PackageReference Include="CommunityToolkit.Mvvm" Version="8.1.0" />
    <PackageReference Include="Dapper" Version="2.0.123" />
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="7.0.0" />
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.3.230202101-experimental1" />
    <PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.25295-preview" />
    <PackageReference Include="System.Data.SQLite.Core" Version="1.0.117" />

As far as I know those are not using Roslyn analyzers/source generators.
I am using VS studio 2022(64-bit)  Version 17.4.5  (Not Preview)

**Comment by michael-hawker** [comment]
@lfhenrl as far as we're aware the only time this occurred was if you were using the `CsWin32` package, it seems like you may be facing a different issue? (Doesn't appear like Dapper or SQLite would use source generators, SQLite 1.0.115.5 would be old enough to confirm that, though I didn't see mention in release notes.)

Can you provide more details about the _exact_ error message you're getting?

FYI @Sergio0694

**Comment by polymorphicshade** [comment]
After playing around with a few things, here are my results so far:

I can reproduce the issue every single time I change anything related to RelayCommand or ObservableProperty (even by simply adding a single Attribute parameter like IncludeCancelCommand).

Editing, deleting, copying, and moving RelayCommands and ObservableProperties permanently breaks Visual Studio's intellisense. The only fix is to close and re-open Visual Studio.

My current configuration:
- Windows 10 x64
- Visual Studio x64 17.5.0
- Resharper 2022.3.2 (enabling or disabling Resharper seems to have no effect on the issue)
- a solution with C# project(s) targeting .NET 7.0
- UI project(s) reference CommunityToolkit.Mvvm 8.1.0
- I am not using CsWin32 in any way
- deleted all Visual Studio and Resharper cache prior to development

For me currently, using CommunityToolkit.Mvvm 8.1.0 in its current state effectively triples my development time.

**Comment by michael-hawker** [comment]
@polymorphicshade if your project still builds and runs and intellisense is breaking, that's a different issue. One which probably should be reported via the tools in Visual Studio. FYI @Sergio0694.

**Comment by polymorphicshade** [comment]
@michael-hawker ah you are right... I see this issue is specific to a build failure. Thanks for the direction.

**Comment by lfhenrl** [comment]
@michael-hawker My problem look like to be the same as @polymorphicshade So I will work on something else until some solution is found in VS-tools or some other place.

**Comment by michael-hawker** [comment]
@lfhenrl thanks for the confirmation, and @polymorphicshade if either of you can repro the issue while the VS feedback log recorder is going and submit an issue to the VS team and link here, we can follow up with the proper folks that work on the source generator tools to try and better understand the issues you're facing.

**Comment by Sergio0694** [comment]
Yeah we need people to submit reports to VS including logs and everything, and also a tracking issue for Roslyn.
It definitely seems like something's up and I don't think it's caused by the Toolkit at all.

**Comment by mobycorp** [comment]
Where are we with this? When I compile my MAUI app, I get a lot of
Analyzer
'CommunityToolkit.Maui.Analyzers.UseCommunityToolkitInitializationAnalyzer'
threw an exception of type 'System.InvalidOperationException' with message
'Wow, this shouldn't happen...' warnings in my output window. Today, I
started to have some odd errors that I can't quite put my finger on, but in
what investigation I could do I found something of interest.

I do not remember installing CsWinRT or CsWin32, but I started to get the
following errors today:

2023-02-27 13:54:16.0779 ERROR    Class: MauiProgram
 Method: CurrentDomain_FirstChanceException exception message:    at
WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
.
2023-02-27 13:54:16.0779 ERROR    Class: MauiProgram
 Method: CurrentDomain_FirstChanceException exception message:
System.InvalidOperationException: Exception of type
'System.InvalidOperationException' was thrown.
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr).
2023-02-27 13:54:23.9592 ERROR    Class: MauiProgram
 Method: CurrentDomain_FirstChanceException exception message:    at
WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
.
2023-02-27 13:54:23.9592 ERROR    Class: MauiProgram
 Method: CurrentDomain_FirstChanceException exception message:
System.ArgumentException: Value does not fall within the expected range.
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr).
2023-02-27 13:54:26.4246 ERROR    Class: MauiProgram
 Method: CurrentDomain_FirstChanceException exception message:    at
WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
.
2023-02-27 13:54:26.4246 ERROR    Class: MauiProgram
 Method: CurrentDomain_FirstChanceException exception message:
System.ArgumentException: Value does not fall within the expected range.
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr).

This is formatted NLog output that was trapped in a FirstChanceException
handler. When I was searching for something that would give me a hint, I
found out that the ThrowExceptionForHR exception is defined in
the Microsoft.Windows.CsWinRT library. Are there two dots that can be
connected here? I am virtually dead in the water now. I did downgrade
my CommunityToolkit.Mvvm from 8.1.0 back to 8.0.0, but that didn't seem to
fix it... I hope this bit helps...

Steve

**Comment by michael-hawker** [comment]
@mobycorp the `CommunityToolkit.Maui.Analyzers.UseCommunityToolkitInitializationAnalyzer` is not part of this repository. You should file an issue on the [MAUI Community Toolkit repo](https://aka.ms/toolkit/maui).

**Comment by mobycorp** [comment]
I mentioned the Maui library by mistake. I meant to say Mvvm… Sorry about
that.

**Comment by mobycorp** [comment]
What is the status of this? I have to restart Visual Studio many, Many, MANY times each day because the code generates fail (sometimes within minutes of each restart) and it has become uber burdensome... Can we get an ETA or status? Should I abandon this NuGet package? Is there a workaround other than restarting VS all the damn time....

**Comment by hymccord** [comment]
@mobycorp Again, this is for Mvvm. Not Maui as the errors you posted also suggest Maui problems.

This GitHub issue is specifically for the **compiler** warning `CS8785`. If you are getting errors then you need to file a new issue with the appropriate details.

**Comment by mobycorp** [comment]
Hank, you need to read my post. I am not complaining about the compiler errors, I am complaining about the analyzers failing. This is an MVVM issue as it has to do with anything marked with the ObservableProperty simply getting lost requiring a VS restart. When the analyzer fails, I agree, the compiler errors get generated, but that isn't why I am posting here... I am complaining about the source generators constantly aborting and screwing everything up.

**Comment by Sergio0694** [comment]
> "This is an MVVM issue as it has to do with anything marked with the ObservablePropert"

If you go through the linked issue (https://github.com/dotnet/roslyn/issues/67123), you'll see this is very likely not actually caused by the MVVM Toolkit at all, it just happens that the MVVM Toolkit analyzers are triggering this issue relatively often, unfortunately.

**Comment by mobycorp** [comment]
Hi Sergio,

First off, I am not trying to be an antagonist here, I am just trying to get my job done. I'll supply you with anything you need, but at the end of the day, I am simply looking for a solution to this issue.

I just read the related issue and I see where you state you are unable to reproduce the error... I can get this to fail at least a dozen times a day... And, while I empathize with you in your belief that this is caused by another project, the fact that I only see this with the MVVM toolkit points me in this direction. If I buy something from Amazon and it doesn't work after it's delivered, I deal with Amazon and not the manufacturer of the defective product. The same applies here. I realize you and the community have worked very hard on this toolkit and it does AMAZING things, but it is broken... From this side of the fence, I only see the MVVM toolkit as the point of failure - you are Amazon in this analogy and the other team (VS/Roslyn ???) are the culprits - I am expecting you to be working with them to get this identified and fixed.

With all of that said, what is being done to coordinate this with the VS Roslyn team? Do they know when a fix will be available? Is going back to v8.0 the best alternative until the real issue gets solved?

Again, I appreciate this package and what it does.

Thanks!
Steve

**Comment by hymccord** [comment]
It's more like Roslyn is Amazon and CommunityToolkit is a storefront using Amazon services to provide you with wares.

> With all of that said, what is being done to coordinate this with the VS Roslyn team? Do they know when a fix will be available? 

Reading the linked issue Sergio provided will answer that and what's being done (or not done).

**Comment by mobycorp** [comment]
At the end of all this, should I downgrade to v8.0 until a fix is in hand? I would be hopeful that the VS restarts could be avoided that way...

**Comment by RodBarnes** [comment]
Just adding my input on this; maybe the results of my investigation will be helpful. NOTE: I originally submitted this an issue to [Developer Community #10291403](https://developercommunity.visualstudio.com/t/Errors-persist-display-even-after-succes/10291403) but, realizing it is only happening with things related to this toolkit, moving here.

For context, I've been experiencing this since v8.0.  Upgrading to  8.1 didn't fix it.  Moving back to has 8.0 fixed it for now but I expect it will reappear after some number of VS sessions since it happened previously in 8.0.

**Comment by codengine** [comment]
I have the same issue from time to time in my project here:
https://github.com/codengine/SOTFEdit

Sometimes it works for a while but stops working from time to time. Usually with the "syntaxTree" issue.

I have the suspicion that it could be related to two reasons:

- Using default values
- Having field names that also represent types (e.g. your ObservableProperty is named _myModel, the generated Property is named MyModel and if there is also a class that is named MyModel would lead to the issue) but haven't investigated it further.

**Comment by l619534951** [comment]
> Sometimes it works for a while but stops working from time to time. Usually with the "syntaxTree" issue.

I have the same issue, but the project can still run

**Comment by codengine** [comment]
> > Sometimes it works for a while but stops working from time to time. Usually with the "syntaxTree" issue.
> 
> I have the same issue, but the project can still run

Yup same for me. Compiles but I get hundreds of errors in VS (I'm using 17.5.2)

**Comment by RodBarnes** [comment]
[This info is found in the Development Community submission but copying it hear for convenience.]

In my case, I can this issue has a couple related fators:

1. The error `The "CompileXAML" task failed unexpectedly.` almost every time -- usually immediately after restart VS following one of these events; it appears for maybe 1-2 seconds and disappears. Based upon that behavior, it appears to be "left over" from the prior session, gets picked up at start up of VS, and then disappears as it "cleans up" or whatever it is doing during startup.  I have seen this error one time during a session, but only once.
2.  In every case I have seen, the issue appears only related to properties/objects with attribute tags from this toolkit or deriving from those classes; e.g., `ObservableProperty`, `ObservableObject`, etc.

**Comment by RodBarnes** [comment]
@mobycorp 
> At the end of all this, should I downgrade to v8.0 until a fix is in hand? I would be hopeful that the VS restarts could be avoided that way...

My experience has been that down-grading to v8.0 does fix it but only briefly.  In a short while (maybe after 3-4 sessions), it begins happening again.

**Comment by RodBarnes** [comment]
@codengine 

> * Having field names that also represent types (e.g. your ObservableProperty is named _myModel, the generated Property is named MyModel and if there is also a class that is named MyModel would lead to the issue) but haven't investigated it further.

Yes, I can confirm that one, too.  When I first noticed that I went through the entire project and renamed every property (of a type from this toolkit) to distinguish the type from the name.  This seemed to work -- maybe reduced the errors a little -- but there are still a ton such that the "Error Window" eventually becomes useless and I have to restart to clear it out.

**Comment by codengine** [comment]
> @codengine
> 
> > * Having field names that also represent types (e.g. your ObservableProperty is named _myModel, the generated Property is named MyModel and if there is also a class that is named MyModel would lead to the issue) but haven't investigated it further.
> 
> Yes, I can confirm that one, too. When I first noticed that I went through the entire project and renamed every property (of a type from this toolkit) to distinguish the type from the name. This seemed to work -- maybe reduced the errors a little -- but there are still a ton such that the "Error Window" eventually becomes useless and I have to restart to clear it out.

I will try to do the same on my project. However, I've used v8.0 yesterday for several hours and haven't had a single error during that time. So at least in my case, this is a valid temporary fix.

**Comment by RodBarnes** [comment]
Well, it seems that some combination of going back to v8.0 and upgrading to the recently-released VS v17.5.3 may've cured this -- at least, I have not seen this behavior in a while.  Anyone else confirm this?

**Comment by codengine** [comment]
Just a small update: I've used version 8.0 for weeks successfully within Rider and got the same issues again within minutes after switching to 8.1 i got the same issues as before. So the issue is independent of the used IDE.

**Comment by Sergio0694** [comment]
Hey everyone! 👋
[Visual Studio 2022 17.6 just released](https://devblogs.microsoft.com/visualstudio/visual-studio-2022-17-6-now-available/)! 🚀
This should include a whole lot of fixes that should mitigate or solve these problems entirely.
Could you try it out and let us know if the issue has been resolved for you now? Thank you! 🙂

**Comment by codengine** [comment]
I updated Visual Studio, upgraded MVVM Toolkit to 8.2.0, cleaned and rebuilt everything and the situation is as bad as before - at least on my end.

**Comment by Sergio0694** [comment]
_\*sigh\*_ 😭

Thank you for sharing though, it helps!

**Comment by sharwell** [comment]
@codengine What is the exact warning message produced by the IDE when you observe the problem? It should say something specific to an exception occurring in the source generator.

**Comment by codengine** [comment]
> @codengine What is the exact warning message produced by the IDE when you observe the problem? It should say something specific to an exception occurring in the source generator.

Maybe I was too quick with my judgement. So here is what I observed.
I went through the Roslyn Analyzer warnings and saw a couple of places where I had [ObservableProperty] on a property and then used the private _property instead of referencing the generated one. This happened a couple of times.

Then there was one place where I had something like this:
Class Name: ScrewStructure
Property: [ObservableProperty] ScrewStructure _screwStructure
I think this example could lead to the code generation to fail as the generated Property name is equal to the class name (ScrewStructure in this example).

To make sure there was no other influence, I restarted my computer, cleaned the whole temp directory, build the solution from scratch and for now, the Analyzer is happy.

I then created a test project which contains the scenario in isolation.

`public partial class TestClass: ObservableObject
{
    [ObservableProperty]
    private AnotherClass _anotherClass;

    public void Foo()
    {
        if (AnotherClass != null)
        {
            
        }
    }
}`

First, Roslyn was not happy (unfortunately Rider is a bit hard to debug in that regard).
Then I renamed the property to "Another", Roslyn was happy again, I renamed it back to "AnotherClass" and Roslyn was still happy.

So in the end, it could have been me who accidentally had the Class Name as the Property Name in the generated source.
I will keep and eye on that and will report back as soon I notice something else.

**Comment by polymorphicshade** [comment]
@Sergio0694 I've updated everything on my end, and I'm no longer seeing the problem! However I think I have to play around with it some more to confirm.

**Comment by Balkoth** [comment]
I'm on 8.2.0 and VS 17.6.1 and the problem is still there.

**Comment by sharwell** [comment]
> I'm on 8.2.0 and VS 17.6.1 and the problem is still there.

@Balkoth can you provide some specific messages you are observing with these version when the errors appear?

**Comment by polymorphicshade** [comment]
@Sergio0694 update: after several days of use, I can confirm the issue is still here.

Over a small period of time, I can code normally and never see this issue. However, some time after that I start seeing the problem again.

Now just like before, as soon as I do anything related to the CommunityToolkit.Mvvm (even as simple as placing `[RelayCommand]` over a private method), intellisense completely breaks in both XAML and CS code.

Still, for me, the only temporary solution to this problem is to clean the solution, close Visual Studio, manually delete **all** my bin and obj folders, and re-launch Visual Studio.

I'm using the latest release version of Visual Studio, CommunityToolkit.Mvvm, and

**Comment by balthasart** [comment]
> @Sergio0694 update: after several days of use, I can confirm the issue is still here.

Same thing for me - however, as all my mvvm related code is on a separate library project - I juste need to "Unload Project" and reload it, then every thing is fine (no need to quit VS, delete bin and obj folders)

**Comment by heartacker** [comment]
the important thing is it stops the hot reload :cry

**Comment by jaredpar** [comment]
Based on the feedback it seems like we addressed part of the problem but not all of it here. Unfortunately the inability to have a reliable repro is making it difficult to track down this issue. It's not even clear right now if there is a bug in the roslyn side or in the mvvm generator. 

One item we're going to try on our end is expand our diagnostic capabilities: 

- https://github.com/dotnet/roslyn/issues/68401
- https://github.com/dotnet/roslyn/issues/68402

Taken together those two issues should let us view the internal mechanics of generators if you submit an ETL trace. That should be integrated into the VS feedback system such that it can be automated to a large degree. 

Our goal is to get this all implemented in time for VS 17.7 Preview 3 or Preview 4. Hopefully that's in time for you all to send us ETL traces with the expanded diagnostics and we can track down where the remaining issues are.

**Comment by Sergio0694** [comment]
@jaredpar I have a (pretty consistent) repro!! 🎉

1. Checkout to https://github.com/jenius-apps/ambie/commit/375e0761d540bb822e7c74e6e819d667ce8d4f02
1. Open the solution (needs the UWP SDK installed as well, just let VS install anything missing)
2. Open `SoundViewModel.cs`
3. Scroll down to `DeleteSound()` (line 373)
4. Delete the line with `[RelayCommand]` over the method
5. Add the attribute back (eg. by manually typing it, or pasting, doesn't seem to matter)

With these steps, I can pretty consistently get the generator to crash. It doesn't _always_ happen though. One time it didn't work, but after closing and reopening VS, doing the same steps did reproduce the issue. Happens almost all the times.

Error message:

> Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Warning	CS8785	Generator 'RelayCommandGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentException' with message 'The hintName 'AmbientSounds.ViewModels.SoundListViewModel.LoadAsync.g.cs' of the added source file must be unique within a generator.
Parameter name: hintName'	AmbientSounds	C:\Users\sergiopedri\git\ambie\src\AmbientSounds\AmbientSounds.csproj	1	Active	Generator threw the following exception:
'System.ArgumentException: The hintName 'AmbientSounds.ViewModels.SoundListViewModel.LoadAsync.g.cs' of the added source file must be unique within a generator.
Parameter name: hintName
   at Microsoft.CodeAnalysis.AdditionalSourcesCollection.Add(String hintName, SourceText source)
   at Microsoft.CodeAnalysis.SourceOutputNode`1.AppendOutputs(IncrementalExecutionContext context, CancellationToken cancellationToken)'.

Also happy to share more from my end if you need (eg. if there's a way to collect more telemetry for you or anything).

> "It's not even clear right now if there is a bug in the roslyn side or in the mvvm generator."

Not excluding there might be a bug in the MVVM Toolkit, but FWIW this repro is using a different generator (the one for `[RelayCommand]`) that is completely different than the one for `[ObservableProperty]` that almost everyone else here and in https://github.com/dotnet/roslyn/issues/67123 was using. I mean I guess it's possible both have the same bug, but I'm kinda inclined to think this might rather be a bug in the generator infrastructure, given the error message is also the same 🤔

Does the repro help?

**Comment by jaredpar** [comment]
@Sergio0694 

> Exception was of type 'ArgumentException' with message 'The hintName 'AmbientSounds.ViewModels.SoundListViewModel.LoadAsync.g.cs' of the added source file must be unique within a generator

That is generally indicative of a bug in the generator though, not roslyn. It's asserting that you provided the same hint name twice. Have you debugged in to make sure you're not doing this?

**Comment by chsienki** [comment]
@jaredpar @Sergio0694 This sounds a lot like a previous issue we had where we were incorrectly handling removed entries: https://github.com/dotnet/roslyn/pull/66643

I would bet that this is probably the same kind of issue.

**Comment by chsienki** [comment]
@Sergio0694 As an aside, I feel like this is the first time we're actually seeing the `Warning CS8785` for this issue, is there a case where VS is not reporting this and just silently failing?

**Comment by Sergio0694** [comment]
Ok I have investigated this a bit, if it's an issue in my generator I have to be honest I just don't see it 🥲
I have however done the following tests as well:
- Changed the `RegisterSourceOutput` to just always use a new `Guid` for filename. Works just fine and no longer repros.
- Changed the incremental model to also have a `Guid` property, which I always set to a new `Guid` in my FAWMN transform. This also works just fine and no longer repros.

So it seems the only way to reproduce the issue (at least, in this case) is if the incremental models compare to be equal across pipelines. I am also a bit confused as to why using a new guid from the output node doesn't result in a build error in case the generator driver was in fact accidentally passing the same model twice (which would explain the `ArgumentException` from the generator) I was seeing. But then wouldn't it have resulted in the same code being generated twice now, which would presumably fail to build (since the generated member would end up being duplicated)? 🤔

> "I feel like this is the first time we're actually seeing the Warning CS8785 for this issue, is there a case where VS is not reporting this and just silently failing?"

Not sure, this is the only case that I can consistently reproduce locally, not sure about others. Hopefully others in this thread that have also encountered the issue (especially with the other generator, for observable properties) can also comment.

**Comment by bad-samaritan** [comment]
I was experiencing `MissingMethodException` for `ObservablePropertyGenerator` in my WPF project.
I tried various workarounds including `<IncludePackageReferencesDuringMarkupCompilation>true` but none of them worked.
Finally I have **upgraded Visual Studio 2022 to 17.6.2 and the error is now gone**.

However, I'm not entirely sure it's gone forever as previously error appeared spontaneously when jumping between different commits of MahApps.Metro (linked as project reference to my app, not nuget) and did not go away after going back to the commit I started with. Cleanups, manually removing bin&obj, restarting VS - nothing helped.

Project details: WPF MVVM app, built on top of MahApps.Metro, targeted to .NET Framework 4.8 & NET 6.0 Windows, View Model with generated properties placed in the same project as main UI (with `<UseWPF>true`).

**Comment by TranquilChaoss** [comment]
Not sure this is of much use, but updating to 17.6.2 from a much older version (17.3.something) solved this issue for me. Am a novice coder and was following the ".NET MAUI Data Binding with MVVM & XAML [5 of 8] | .NET MAUI for Beginners" youtube tutorial from the dotnet team and got this issue. Will leave a comment about this solution on the YouTube video as well as the option to downgrade to 8.0. Sadly didn't think to document the 60+ errors and warnings (which seems like more than others getting the same issue) I was getting on 8.2, but outside of using the 8.0 version the build would fail and the source code generators would not generate any files.

**Comment by Sergio0694** [comment]
The last issues related to this have been addressed by https://github.com/dotnet/roslyn/pull/68451.
Fixes should flow downstream from VS 17.7 Preview 4. Closing this 🙂

**Comment by DanielD** [comment]
It's still happening on Visual Studio for Mac version 17.6 (build 1575).

**Comment by sharwell** [comment]
@DanielD You can write the following in a source file to see the compiler version in use:

```
#error version
```

The fix would only be in Roslyn 4.7 and higher.

**Comment by LiamJM96** [comment]
@Sergio0694 Thanks for the update and quick response 🙂

**Comment by CodingOctocat** [comment]
Its still happening on WPF + .NET 7.0 + VS 17.6.5 + MvvmToolkit v8.2.1,.

**Comment by sharwell** [comment]
@mscholz5 That does not sound like the same issue. Can you file a new issue?

**Comment by mscholz5** [comment]
I filed #750.

**Comment by LancashireDave** [comment]
This issue is also happening for me on 17.8.0 Preview 7.0 and 8.2.2.
